synthtool/__main__.py

```diff
@@ -67,35 +84,40 @@ def main(synthfile: str, metadata: str, extra_args: Sequence[str]):
     f"""Synthesizes source code according to the instructions in synthfile arg.
 
     Optional environment variables:
-      SYNTHTOOL_ARTMAN_VERSION:  The version of artman to use.
-      SYNTHTOOL_GOOGLEAPIS:      Path to local clone of https://github.com/googleapis/googleapis
-      SYNTHTOOL_GENERATOR:       Path to local gapic-generator directory to use for generation.
+    SYNTHTOOL_ARTMAN_VERSION:  The version of artman to use.
+    SYNTHTOOL_GOOGLEAPIS:      Path to local clone of https://github.com/googleapis/googleapis
+    SYNTHTOOL_GENERATOR:       Path to local gapic-generator directory to use for generation.
                 By default, the latest version of gapic-generator will be used.
-      AUTOSYNTH_USE_SSH:         Access github repos via ssh instead of https.
-      {preconfig.PRECONFIG_ENVIRONMENT_VARIABLE}:  Path to a json file.
+    AUTOSYNTH_USE_SSH:         Access github repos via ssh instead of https.
+    {preconfig.PRECONFIG_ENVIRONMENT_VARIABLE}:  Path to a json file.
 
 
     {preconfig.PRECONFIG_HELP}
     """
-    _extra_args.extend(extra_args)
+    if synthfile == "template":
```

Comment:
I'm not sure how this works with click and executing as a module, but this feels wrong.

Perhaps we should add this feature as `python3 -m synthtool.templates` rather than overloading the main module execution.

Reply:
So, add a `__main__.py` to templates?

I was trying to figure out how to make click handle a default command plus other commands, and it wasn't immediately obvious.

Reply:
I agree that we should put this command in its own .py file, and invoke it directly from there.

I like to see verbs in command names.  What is synthtool doing with the templates?  Create a file <verb_templates>.py with its own main function and move this code in there.